uniflow/op/model/model_server.py

```diff
@@ -769,3 +770,223 @@ def __call__(self, data: List[str]) -> List[str]:
+                    "Please use a profile instead."
+                )
+            else:
+                session = boto3.Session(profile_name="default")
+                warnings.warn(
+                    "Using default profile to create the session. "
+                    "Please pass the profile name in the model config."
+                )
+
+            aws_region = (
+                self._model_config.aws_region if self._model_config.aws_region else None
+            )
+
+            self._client = session.client("sagemaker-runtime", region_name=aws_region)
+
+        except ImportError as e:
+            raise ModuleNotFoundError(
+                "Failed to import the 'boto3' Python package. "
+                "Please install it by running `pip install boto3`."
+            ) from e
+        except Exception as e:
+            raise ValueError(
+                "Failed to load credentials for authenticating with the AWS client. "
+                "Please ensure that the specified profile name contains valid credentials."
+            ) from e
+
+    def _preprocess(self, data: List[str]) -> List[str]:
+        """Preprocess data.
+
+        Args:
+            data (List[str]): Data to preprocess.
+
+        Returns:
+            List[str]: Preprocessed data.
+        """
+        return data
+
+    def _postprocess(self, data: List[str]) -> List[str]:
+        """Postprocess data.
+
+        Args:
+            data (str): Data to postprocess.
+
+        Returns:
+            List[str]: Postprocessed data.
+        """
+        return data
+
+    def enforce_stop_tokens(self, text: str, stop: List[str]) -> str:
+        """Cut off the text as soon as any stop words occur."""
+        return re.split("|".join(stop), text, maxsplit=1)[0]
+
+    def prepare_input(
+        self, model_type: str, prompt: str, model_kwargs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Prepare the input for the model based on the model_type.
+
+        Args:
+            model_type (str): The type of the model.
+            prompt (str): The input prompt.
+            model_kwargs (Dict[str, Any]): Additional model arguments.
+
+        Returns:
+            Dict[str, Any]: The prepared input for the model.
+        """
+
+        def prepare_falcon_input(
+            prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
+            input_body = {
+                "inputs": f"{prompt}",
+                "parameters": model_kwargs,
+            }
+            return json.dumps(input_body)
+
+        def prepare_default_input(
+            prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
+            input_str = json.dumps({"inputs": prompt, "parameters": model_kwargs})
+            return input_str
+
+        model_input_preparation = {"falcon": prepare_falcon_input}
+
+        prepare_input_for_model = model_input_preparation.get(
+            model_type, prepare_default_input
+        )
```

Comment:
nit: I remember there is also `model_input_preparation`-ish thing for bedrock endpoint. Is it possible that we can unified them into a class for re-usability.